ps_checkout.php

```diff
@@ -1039,20 +1039,77 @@ public function hookDisplayAdminOrderMainBottom(array $params)
      */
     public function hookDisplayPaymentTop()
     {
-        if (false === $this->context->cookie->__isset('paypalOrderId')) {
-            return '';
+        $paymentError = (int) Tools::getValue('paymentError');
+        $paymentErrorMessage = '';
+        $isExpressCheckout = $this->context->cookie->__isset('paypalOrderId');
+
+        if (0 < $paymentError) {
+            switch ($paymentError) {
+                case \PrestaShop\Module\PrestashopCheckout\Exception\PsCheckoutException::PAYPAL_PAYMENT_CARD_ERROR:
```

Comment:
use a namespace here? :)

Reply:
We cannot due to support of PrestaShop 1.6, there were an eval() cause fatal error.